src/Config/Adapter/Ini.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+/**
+ * This file is part of the Phalcon Framework.
+ *
+ * (c) Phalcon Team <team@phalcon.io>
+ *
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Phalcon\Config\Adapter;
+
+use Phalcon\Config\Config;
+use Phalcon\Config\Exception;
+
+use function basename;
+use function count;
+use function is_array;
+use function is_numeric;
+use function is_string;
+use function parse_ini_file;
+use function preg_match;
+use function strpos;
+use function strtolower;
+use function substr;
+
+use const INI_SCANNER_RAW;
+
+/**
+ * Reads ini files and converts them to Phalcon\Config objects.
+ *
+ * Given the next configuration file:
+ *
+ *```ini
+ * [database]
+ * adapter = Mysql
+ * host = localhost
+ * username = scott
+ * password = cheetah
+ * dbname = test_db
+ *
+ * [phalcon]
+ * controllersDir = "../app/controllers/"
+ * modelsDir = "../app/models/"
+ * viewsDir = "../app/views/"
+ * ```
+ *
+ * You can read it as follows:
+ *
+ *```php
+ * use Phalcon\Config\Adapter\Ini;
+ *
+ * $config = new Ini("path/config.ini");
+ *
+ * echo $config->phalcon->controllersDir;
+ * echo $config->database->username;
+ *```
+ *
+ * PHP constants may also be parsed in the ini file, so if you define a constant
+ * as an ini value before calling the constructor, the constant's value will be
+ * integrated into the results. To use it this way you must specify the optional
+ * second parameter as `INI_SCANNER_NORMAL` when calling the constructor:
+ *
+ * ```php
+ * $config = new \Phalcon\Config\Adapter\Ini(
+ *     "path/config-with-constants.ini",
+ *     INI_SCANNER_NORMAL
+ * );
+ * ```
+ */
+class Ini extends Config
+{
+    /**
+     * Ini constructor.
+     *
+     * @param string   $filePath
+     * @param int|null $mode
+     *
+     * @throws Exception
+     */
+    public function __construct(string $filePath, $mode = null)
+    {
+        // Default to INI_SCANNER_RAW if not specified
+        if (null === $mode) {
+            $mode = INI_SCANNER_RAW;
+        }
+
+        $iniConfig = parse_ini_file($filePath, true, $mode);
```

Comment:
Utilize `Phalcon\Support\Traits` and create a new one to wrap `parse_ini_file` as `phpParseIniFile`. This way we can stub that protected method and force the exception in a test. Same thing with the YAML extension (`yaml_parse_file`) as well as `extension_loaded`

Reply:
If you rebase with v5, you will see a `phpFunctionTrait` in `Support\Traits`. We can utilize that trait for the functions above.

Reply:
Added `parse_ini_file` to `PhpFileTrait` and added `extension_loaded` to `PhpFunctionTrait`.
Created a dedicated `PhpYamlTrait` for the `yaml_parse_file` to avoid import problems when the `yaml` extension is not loaded.